apps/tangle-cloud/src/pages/blueprints/[id]/deploy/DeploySteps/components/AssetRequirementFormItem.tsx

```diff
@@ -1,90 +1,97 @@
 import { RestakeAssetId } from '@tangle-network/tangle-shared-ui/types';
-import { Input, Typography } from '@tangle-network/ui-components';
+import { Typography, Label, Slider } from '@tangle-network/ui-components';
 import { FC } from 'react';
-import { PrimitiveAssetMetadata } from '@tangle-network/tangle-shared-ui/types/restake';
 import LsTokenIcon from '@tangle-network/tangle-shared-ui/components/LsTokenIcon';
 import ErrorMessage from '../../../../../../components/ErrorMessage';
 import cx from 'classnames';
+import { AssetSchema } from '../../../../../../utils/validations/deployBlueprint';
+import { LabelClassName } from '../type';
 
-type AssetRequirementFormItemProps = {
-  index: number;
+type BaseAssetRequirementFormItemProps = {
   assetId?: RestakeAssetId;
-  onChangeMinExposurePercent: (value: string) => void;
-  onChangeMaxExposurePercent: (value: string) => void;
-  assetMetadata?: PrimitiveAssetMetadata | null;
-  minExposurePercent?: string;
-  maxExposurePercent?: string;
+  assetMetadata?: AssetSchema | null;
+  minExposurePercent?: number;
+  maxExposurePercent?: number;
+  className?: string;
+};
+
+type ViewOnlyAssetRequirementFormItemProps = BaseAssetRequirementFormItemProps;
+
+type AssetRequirementFormItemProps = BaseAssetRequirementFormItemProps & {
+  onChangeExposurePercent: (value: number[]) => void;
   minExposurePercentErrorMsg?: string;
   maxExposurePercentErrorMsg?: string;
-  className?: string;
 };
 
-export const AssetRequirementFormItem: FC<AssetRequirementFormItemProps> = ({
-  index,
-  assetId,
-  minExposurePercent,
-  maxExposurePercent,
-  onChangeMinExposurePercent,
-  onChangeMaxExposurePercent,
-  minExposurePercentErrorMsg,
-  maxExposurePercentErrorMsg,
-  assetMetadata,
-  className,
-}) => {
+export const AssetRequirementFormItem: FC<
+  AssetRequirementFormItemProps | ViewOnlyAssetRequirementFormItemProps
+> = (props) => {
+  const {
+    assetId,
+    minExposurePercent,
+    maxExposurePercent,
+    assetMetadata,
+    className,
+  } = props;
+
+  const isViewOnly = !('onChangeExposurePercent' in props);
```

Comment:
In case prop name changes in the future:
```suggestion
  const isViewOnly = !(('onChangeExposurePercent' satisfies keyof AssetRequirementFormItemProps | ViewOnlyAssetRequirementFormItemProps) in props);
```